Bioinfo_joint/HW4_modules/natls.py

```diff
@@ -0,0 +1,110 @@
+def dict_comp(nucleotide: str) -> str:
+    complement = dict(
+        [
+            ("a", "t"),
+            ("A", "T"),
+            ("c", "g"),
+            ("C", "G"),
+            ("g", "c"),
+            ("G", "C"),
+            ("t", "a"),
+            ("T", "A"),
+            ("u", "a"),
+            ("U", "A"),
+            ("0", "0"),
+        ]
+    )

```

Comment:
Использование словаря - это очень хорошая идея, но в данном случае, словарь определяется при каждом вызове этой функции, что неоптимально. Определение словаря можно вынести на самый верхний глобальный для файла уровень видимости без отступа. В таком случае, он не должен определяться при каждом вызове функции.

Также, его можно определить более легковесным способом с точки зрения синтаксиса.

```suggestion
    complement = {
            "a": "t",
            "A": "T",
            "c": "g",
            "C": "G",
            "g": "c",
            "G": "C",
            "t": "a",
            "T": "A",
            "u": "a",
            "U": "A",
            "0": "0"
      }
```